server/content-repo/content-delivery.router.js

```diff
@@ -0,0 +1,40 @@
+const { ContentRepo, Enrollment } = require('../common/database');
+const { createError } = require('../common/errors');
+const ctrl = require('./content-delivery.controller');
+const HttpStatus = require('http-status');
+const router = require('express').Router();
+
+const { NOT_FOUND, UNAUTHORIZED } = HttpStatus;
+
+router
+  .use('/*', hasAccess)
+  .get('/', ctrl.list)
+  .use('/:contentId*', getRepo)
+  .get('/:contentId', ctrl.get)
+  .get('/:contentId/container/:containerId', ctrl.getContainer)
+  .get('/:contentId/exam/:examId', ctrl.getExam);
+
+function hasAccess({ cohort, user }, res, next) {
+  if (user.isAdmin()) return next();
+  const opts = { where: { studentId: user.id, cohortId: cohort.id } };
+  return Enrollment.count(opts)
+    .then(count => {
+      if (count) return next();
+      return createError(UNAUTHORIZED, 'Access restricted');
+    });
+}
+
+function getRepo(req, res, next) {
```

Comment:
Fetching repo when user is trying to get container might be too big performance penalty, therefore I would consider not loading it on the context if its not needed. Since this is content delivery, we should focus on performance (this will be used a lot).

Reply:
Yes, You are right. I did this to check if ContentRepo is in that Cohort. Then I would suggest that our path for saving imported content use ContentRepoId instead of SourceId so we ensure nobody can fetch wrong content (SourceId can be thesame in different cohorts, but the content can be out of sync).

Reply:
Also, another reason for fetching repo on content delivery is to get it's sourceId (because of our imported path), so I think we definitely need to use ContentRepoId for that.